### PR TITLE
feat: Added ENABLE_COORDINATOR_ORDER_CREATE flag

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
Added ENABLE_COORDINATOR_ORDER_CREATE flag
* calling commerce coordinator Order create API.

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/extensions/checkout/mixins.py [hunk 1]
```diff
@@ -93,6 +95,18 @@ def add_payment_event(self, event):  # pylint: disable = arguments-differ
...
 
+    def handle_commerce_order_create(self, basket):
+        """
+        Handle commerce coordinator order creation.
+
+        Calls create_commerce_coordinator_order if flag is active and basket is a non-program purchase.
+        """
+        request = crum.get_current_request()
+        if waffle.flag_is_active(request, ENABLE_COORDINATOR_ORDER_CREATE):
+            is_seat_purchase = all(line for line in basket.lines.all() if line.product.is_seat_product)
```
Comment (HammadAhmadWaqas) [inline ecommerce/extensions/checkout/mixins.py]: I just got [confirmation](https://twou.slack.com/archives/C03S3D8736H/p1678441566585479) that non-program orders/baskets should contain only one seat product, So I think we can simplify the logic here. I will be pushing the updated code soon.

File: ecommerce/extensions/checkout/mixins.py [hunk 2]
```diff
@@ -136,6 +152,28 @@ def emit_checkout_step_events(self, basket, handled_processor_response, payment_
...
+        Calls /ecommerce/order API from commerce coordinator.
+        """
+        site_config = basket.site.siteconfiguration
+        api_client = site_config.oauth_api_client
+        api_url = site_config.build_commerce_coordinator_url('/ecommerce/order/')
+        query_params = {
+            'edx_lms_user_id': basket.owner.id,
+            'email': basket.owner.email,
+            'product_sku': line.product.stockrecords.first().partner_sku,
+            'coupon_code': list(basket.vouchers.values_list('code', flat=True)),
```
Comment (pshiu) [inline ecommerce/extensions/checkout/mixins.py:166]: We can definitely table this for later, but is `coupon_code` set up to accept more than one coupon on the Coordinator-side?

**Review by pshiu — APPROVED** [review]
Fantastic work, @HammadAhmadWaqas. Thank you!

File: ecommerce/extensions/checkout/mixins.py (hunk 2 above)
Comment (HammadAhmadWaqas) [reply ecommerce/extensions/checkout/mixins.py:166]: Not at the moment. I assume we will allow adding multiple coupons in the coordinator as it is allowed now in ecommerce.

File: ecommerce/extensions/checkout/mixins.py [hunk 3]
```diff
@@ -136,6 +152,28 @@ def emit_checkout_step_events(self, basket, handled_processor_response, payment_
...
 
+    def create_commerce_coordinator_order(self, basket, line):
+        """
+        Calls /ecommerce/order API from commerce coordinator.
+        """
+        site_config = basket.site.siteconfiguration
+        api_client = site_config.oauth_api_client
+        api_url = site_config.build_commerce_coordinator_url('/ecommerce/order/')
+        query_params = {
+            'edx_lms_user_id': basket.owner.id,
```
Comment (pshiu) [inline ecommerce/extensions/checkout/mixins.py:163]: (Note from the future: I missed this - this should be the LMS user id, not the Ecommerce user id.)

**Pull request merged**